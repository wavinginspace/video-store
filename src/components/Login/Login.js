import React, { Component } from 'react';
import './Login.scss';
import { Button, Input, Required } from '../Utils/Utils';
import { Link } from 'react-router-dom';

export class Login extends Component {

  state = { error: null };

  render() {
    const { error } = this.state;
    return (
      <>
        <form className="RegistrationForm box fadeIn" onSubmit={this.handleSubmit}>
          <div role="alert">{error && <p className="red">{error}</p>}</div>

          <div className="user_name login-input">
            <label htmlFor="RegistrationForm__user_name">
              User name <Required />
            </label>
            <Input
              name="user_name"
              type="text"
              autoComplete="off"
              required
              id="RegistrationForm__user_name"></Input>
          </div>
          <div className="password login-input">
            <label htmlFor="Login__password">
              Password <Required />
            </label>
            <Input
              name="password"
              type="password"
              autoComplete="off"
              required
              id="Login__password"></Input>
          </div>
          {/* <div className="password2 register-input">
            <label htmlFor="Login__password2">
              Re-enter Password <Required />
            </label>
            <Input
              name="password2"
              type="password"
              autoComplete="off"
              required
              id="Login__password2"></Input>
          </div> */}

          <Button className=" register-button submit-button" type="submit">
            Login
          </Button>
        </form>

        <Link className="back-button" to="/">
          <p className="back-button-p">Back</p>
        </Link>
      </>
    );
  }
}

export default Login;
