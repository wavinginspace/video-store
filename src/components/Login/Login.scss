@import '../../styles/_breakpoints.scss';
@import '../../styles/_variables.scss';

.Login {
  margin: 0 auto;
}

// .login-form {
//   display: flex;
//   justify-content: center;
//   flex-direction: column;
// }

h2 {
  text-align: center;
}

input, .Input {
  width: 150px;
  height: 1.2rem;
}

.login-input {
  display: flex;
  flex-direction: column;
  align-items: center;
}

#Login__password {
  margin-bottom: 2rem;
}

.red {
  color: red;
}

