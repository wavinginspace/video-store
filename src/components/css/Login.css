.Login {
  margin: 0 auto;
}

.login-form {
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
      -ms-flex-pack: center;
          justify-content: center;
  -webkit-box-orient: vertical;
  -webkit-box-direction: normal;
      -ms-flex-direction: column;
          flex-direction: column;
}

h2 {
  text-align: center;
}
/*# sourceMappingURL=Login.css.map */