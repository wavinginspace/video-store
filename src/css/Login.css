:root {
  --main: var(--main);
  --nav: #499acc;
  --headergrad: linear-gradient(0deg, #ff7676, #fffd8f);
  --light: #fffd8f;
  --headerog: #ff7676;
}

.Login {
  margin: 0 auto;
}

.login-form {
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
      -ms-flex-pack: center;
          justify-content: center;
  -webkit-box-orient: vertical;
  -webkit-box-direction: normal;
      -ms-flex-direction: column;
          flex-direction: column;
}

h2 {
  text-align: center;
}
/*# sourceMappingURL=Login.css.map */