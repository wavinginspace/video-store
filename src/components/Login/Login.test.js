import React from 'react';
import ReactDOM from 'react-dom';
import Login from './Login';
import renderer from 'react-test-renderer';

it('renders without crashing', () => {
  // first create a DOM element to render the component into
  const div = document.createElement('div');

  // render the component, this is the actual test, if something is wrong it will fail here
  ReactDOM.render(<Login />, div);

  // clean up code
  ReactDOM.unmountComponentAtNode(div);
});

it('renders the UI as expected', () => {
  const tree = renderer.create(<Login />).toJSON();
  expect(tree).toMatchSnapshot();
});
