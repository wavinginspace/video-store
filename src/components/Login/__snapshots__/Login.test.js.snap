// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the UI as expected 1`] = `
<div
  className="Login box"
>
  <form
    action=""
    className="login-form"
    name="login-form"
  >
    <label
      htmlFor="email"
    >
      Email:
    </label>
    <input
      className="login-input"
      name="email"
      type="email"
    />
    <label
      htmlFor="email"
    >
      Password:
    </label>
    <input
      className="login-input"
      name="loginpassword"
      type="password"
    />
    <button
      className="login button"
      type="submit"
    >
      Login
    </button>
  </form>
</div>
`;
